src/cf_cre.erl

```diff
@@ -248,31 +249,33 @@ when is_atom( Mod ), is_map( LibMap ) ->
   gen_server:start_link( {local, cre}, ?MODULE, {Mod, ModArg, LibMap}, [] ).
 
 
--spec submit( App, Cwd ) -> cf_sem:fut()
+-spec submit( Runtime, App, Cwd ) -> cf_sem:fut()
 when App    :: cf_sem:app(),
-     Cwd    :: string().
+     Cwd    :: string(),
+     Runtime :: atom() | pid().
 
-submit( App, Cwd )
+submit( Runtime, App, Cwd )
 when is_tuple( App ), is_list( Cwd ) ->
-  gen_server:call( cre, {submit, App, Cwd} ).
+  gen_server:call( Runtime, {submit, App, Cwd} ).
 
 %% =============================================================================
 %% Internal Functions
 %% =============================================================================
 
-stage( Lam, Fa, DataDir, Mod, R, LibMap, ModState )
-when is_tuple( Lam ),
+stage( Runtime, Lam, Fa, DataDir, Mod, R, LibMap, ModState )
+when is_pid( Runtime); is_atom( Runtime ),
```

Comment:
hmm wait a sec. According to this http://erlang.org/doc/reference_manual/expressions.html#id83710
my expression is wrong